Design/Algorithm_ease_of_use.md

```diff
@@ -0,0 +1,88 @@
+Motivation
+==========
+
+Currently code (especially workflow algorithms) will be full of things
+like
+
+```C
+double tolerance = getProperty("CompressTolerance");
+m_outputEW = boost::dynamic_pointer_cast<EventWorkspace>(m_outputW);
+if ((m_outputEW) && (tolerance > 0.)) {
+  g_log.information() << "running CompressEvents(Tolerance=" << tolerance
+                      << ")\n";
+  API::IAlgorithm_sptr compressAlg = createChildAlgorithm("CompressEvents");
+  compressAlg->setProperty("InputWorkspace", m_outputEW);
```

Comment:
That is *really* what I was trying to figure out how to propose. If you have a way to do it, add it in as "Alternative 3".